packages/ui/src/ui/components/ColumnHeader/ColumnHeader.tsx

```diff
@@ -100,3 +171,37 @@ export default function ColumnHeader<T extends string = string>(props: ColumnHea
         </div>
     );
 }
+
+function SubColumnSelector<T extends string>({
+    order,
+    options,
+    onSort,
+}: Pick<ColumnHeaderProps<T>, 'onSort'> & {
+    order?: ColumnHeaderProps<T>['order'];
+    options?: Array<ColumnInfo<T>>;
+}) {
+    return !options ? null : (
+        <DropdownMenu
+            items={options.map((x) => {
+                const {titleContent} = getNameTitle(x);
+                return {
+                    text: titleContent,
+                    action() {
+                        const {column, allowUnordered, withUndefined, allowedOrderTypes} = x;
+                        const nextOrder = allowedOrderTypes
```

Comment:
is it worth doing all these calculations if "onSort" may turn out to be "undefined"?

Reply:
`SubColumnSelector` is displayed when `ColumnHeader.props.options` is defined and there is no reason to have `options` without `onSort`.